Automattic/studio
# Studio: Add UI for chat offline mode

Related to https://github.com/Automattic/dotcom-forge/issues/7818

## Proposed Changes

This PR handles the UI for the offline mode in the assistant.
 
<img width="1057" alt="Screenshot 2024-06-19 at 9 49 40 AM" src="https://github.com/Automattic/studio/assets/25575134/98c345b7-c203-4404-80ad-79097042071c">

<img width="1047" alt="Screenshot 2024-06-19 at 9 49 31 AM" src="https://github.com/Automattic/studio/assets/25575134/85264b98-ea41-4cdc-a3d0-8469c848b40e">


## Testing Instructions

* Pull the changes from this branch locally
* Start the app with the AI Assistant: `STUDIO_AI=true npm start`
* Navigate to the AI Assistant

**Test case 1 - a site with some messages that were previously sent**
* Make sure to send some messages first
* Disable your internet connection (in my case, I disabled WiFi)
* Observe that you can see the offline message and that the input field is disabled
* Confirm that the messages that you sent previously are still displayed above the offline mode message

**Test case 1 - a site with no messages**
* Access the site that has no messages and confirm you can see the Welcome component
* Disable your internet connection (in my case, I disabled WiFi)
* Confirm that the welcome component is not displayed anymore and that you can see only the offline mode message
* Confirm that the input field is disabled

**PLEASE NOTE**:
* This PR does not style the AI input component for when it is disabled - this is handled under https://github.com/Automattic/studio/pull/266 and once that is merged, the AI input will also be styled according to the [design](https://www.figma.com/design/RToz6tIuQ7nlZrikBte4GU/Studio?node-id=3274-115968&t=2tUz1Lufo2Kmr5he-0)
* This PR only adds the Offline mode and the conditions under which it should be displayed
* You might notice that in the designs, the messages flow from bottom to top (as opposed from top to bottom as it is now in the PR and on trunk; this will also be handled in https://github.com/Automattic/studio/pull/266 )
## Pre-merge Checklist
